src/o365/spo/commands/theme/theme-apply.ts

```diff
@@ -51,27 +69,59 @@ class SpoThemeApplyCommand extends SpoCommand {
           cmd.log(`Applying theme ${args.options.name} to the ${args.options.webUrl} site...`);
         }
 
-        const requestOptions: any = {
-          url: `${auth.site.url}/_vti_bin/client.svc/ProcessQuery`,
-          headers: {
-            authorization: `Bearer ${auth.service.accessToken}`,
-            'X-RequestDigest': res.FormDigestValue
-          },
-          body: `<Request AddExpandoFieldTypeSuffix="true" SchemaVersion="15.0.0.0" LibraryVersion="16.0.0.0" ApplicationName="${config.applicationName}" xmlns="http://schemas.microsoft.com/sharepoint/clientquery/2009"><Actions><ObjectPath Id="10" ObjectPathId="9" /><Method Name="SetWebTheme" Id="11" ObjectPathId="9"><Parameters><Parameter Type="String">${Utils.escapeXml(args.options.name)}</Parameter><Parameter Type="String">${Utils.escapeXml(args.options.webUrl)}</Parameter></Parameters></Method></Actions><ObjectPaths><Constructor Id="9" TypeId="{268004ae-ef6b-4e9b-8425-127220d84719}" /></ObjectPaths></Request>`
-        };
+        let requestOptions: any;
+
+        if (isSharePointTheme) {
+          const requestBody: any = this.getSharePointTheme(args.options.name);
+
+          requestOptions = {
+            url: `${args.options.webUrl}/_api/ThemeManager/ApplyTheme`,
+            method: 'POST',
```

Comment:
We already use `request.post` to issue the request so there is no need to specify the method explicitly here.